# Relatório de Vendas
    Foi solicitada uma análise das vendas da empresa nas cidades em que vendemos nossos produtos, precisamos enviar um
    relatório do total de vendas em cada cidade e qual foi a loja com o maior número de arrecadação e qual teve o menor.
    
## Passo a passo do projeto:
    Passo 1: Importar a biblioteca e as bases de dados;
    Passo 2: Unificar as bases de dados em uma só;
    Passo 3: Ajuste e tratamento de dados(caso necessário);
    Passo 4: Extrair o total de vendas em cada cidade;
    Passo 5: Determinar qual a cidade com o maior e o menor número de vendas;
    Passo 6: Escrever o relatório final.

#### Passo 1: Importar a biblioteca e as bases de dados;

In [1]:
import pandas as pd

bd1 = pd.read_excel('Aracaju.xlsx')
bd2 = pd.read_excel('Fortaleza.xlsx')
bd3 = pd.read_excel('Natal.xlsx')
bd4 = pd.read_excel('Recife.xlsx')
bd5 = pd.read_excel('Salvador.xlsx')

Cidade       Data  Vendas  LojaID  Qtde
0    Aracaju 2018-01-01  142.00    1520     1
1    Aracaju 2018-01-01   14.21    1522     6
2    Aracaju 2018-01-01   71.55    1520     1
3    Aracaju 2018-01-01    3.01    1521     7
4    Aracaju 2018-01-01   24.51    1522     8
..       ...        ...     ...     ...   ...
125  Aracaju 2018-01-01   15.62    1522     2
126  Aracaju 2018-01-01   13.41    1520     7
127  Aracaju 2018-01-01   33.12    1522     9
128  Aracaju 2018-01-01   37.49    1521     2
129  Aracaju 2018-01-01   13.70    1522     6

[130 rows x 5 columns]

#### Passo 2: Unificar as bases de dados em uma tabela;

In [4]:
tabela_vendas = pd.concat([bd1, bd2, bd3, bd4, bd5])
display(tabela_vendas)

Cidade       Data  Vendas  LojaID  Qtde
0     Aracaju 2018-01-01  142.00    1520     1
1     Aracaju 2018-01-01   14.21    1522     6
2     Aracaju 2018-01-01   71.55    1520     1
3     Aracaju 2018-01-01    3.01    1521     7
4     Aracaju 2018-01-01   24.51    1522     8
..        ...        ...     ...     ...   ...
235  Salvador 2019-01-03   41.84    1034     1
236  Salvador 2019-01-03  126.29    1035     3
237  Salvador 2019-01-03   38.06    1036     3
238  Salvador 2019-01-03  139.64    1035     1
239  Salvador 2019-01-03  161.41    1037     3

[894 rows x 5 columns]

#### Passo 3: Ajuste e tratamento de dados;

In [7]:
# Vamos verificar os tipos de dados de cada coluna;

tabela_vendas.dtypes

Cidade            object
Data      datetime64[ns]
Vendas           float64
LojaID             int64
Qtde               int64
dtype: object

In [8]:
# A coluna LojaID é um tipo inteiro. Por mais que sejam valores numéricos, não realizaremos operações com eles. 
# Vamos alterá-los para o tipo object.

tabela_vendas['LojaID'] = tabela_vendas['LojaID'].astype('object')
tabela_vendas.dtypes

Cidade            object
Data      datetime64[ns]
Vendas           float64
LojaID            object
Qtde               int64
dtype: object

In [11]:
# Verificar se há volores nulos na tabela;

tabela_vendas.isnull().sum() # Este código retorna a soma da quantidade de valores nulos em cada linha da tabela.

Cidade    0
Data      0
Vendas    0
LojaID    0
Qtde      0
dtype: int64

#### Analisando mais profundamente, não está muito claro o que a coluna de Vendas e Qtde querem dizer exatamente. Também vamos criar uma nova coluna do valor total de vendas.

In [30]:
# Renomeando das colunas Vendas e Qtde;

tabela_vendas.rename(columns={'Vendas':'Valor_Vendas', 'Qtde': 'Qtde_Produtos'}, inplace=True)
display(tabela_vendas)

Cidade       Data  Valor_Vendas LojaID  Qtde_Produtos  Total_Vendas
0     Aracaju 2018-01-01        142.00   1520              1        142.00
1     Aracaju 2018-01-01         14.21   1522              6         85.26
2     Aracaju 2018-01-01         71.55   1520              1         71.55
3     Aracaju 2018-01-01          3.01   1521              7         21.07
4     Aracaju 2018-01-01         24.51   1522              8        196.08
..        ...        ...           ...    ...            ...           ...
235  Salvador 2019-01-03         41.84   1034              1         41.84
236  Salvador 2019-01-03        126.29   1035              3        378.87
237  Salvador 2019-01-03         38.06   1036              3        114.18
238  Salvador 2019-01-03        139.64   1035              1        139.64
239  Salvador 2019-01-03        161.41   1037              3        484.23

[894 rows x 6 columns]

In [31]:
# Criando a coluna Total_Vendas

tabela_vendas['Total_Vendas'] = tabela_vendas["Valor_Vendas"].mul(tabela_vendas["Qtde_Produtos"])
display(tabela_vendas)

Cidade       Data  Valor_Vendas LojaID  Qtde_Produtos  Total_Vendas
0     Aracaju 2018-01-01        142.00   1520              1        142.00
1     Aracaju 2018-01-01         14.21   1522              6         85.26
2     Aracaju 2018-01-01         71.55   1520              1         71.55
3     Aracaju 2018-01-01          3.01   1521              7         21.07
4     Aracaju 2018-01-01         24.51   1522              8        196.08
..        ...        ...           ...    ...            ...           ...
235  Salvador 2019-01-03         41.84   1034              1         41.84
236  Salvador 2019-01-03        126.29   1035              3        378.87
237  Salvador 2019-01-03         38.06   1036              3        114.18
238  Salvador 2019-01-03        139.64   1035              1        139.64
239  Salvador 2019-01-03        161.41   1037              3        484.23

[894 rows x 6 columns]

#### Passo 4: Extrair o total de vendas em cada cidade;

In [37]:
tabela_vendas.groupby('Cidade')['Total_Vendas'].sum()

Cidade
Aracaju       48748.25
Fortaleza     37913.97
Natal        167227.52
Recife        51936.51
Salvador      40596.73
Name: Total_Vendas, dtype: float64

#### Passo 5: Determinar qual a cidade com o maior e o menor número de vendas;

In [42]:
tabela_vendas.groupby('LojaID')['Total_Vendas'].sum()

LojaID
852     52744.00
853     42469.00
854     46655.00
980     13598.62
981     11748.87
982     15819.33
983     23415.17
1002     7983.09
1003     6731.86
1004     4261.30
1005     6292.24
1034     8463.75
1035    18693.44
1036    22235.61
1037    16563.45
1520    12787.81
1521     9421.48
1522    15103.92
1523    11435.04
Name: Total_Vendas, dtype: float64

#### Passo 6: Escrever o Relatório

Prezados, bom dia, segue o relatório do valor total das vendas por cidade:

- Natal        167227.52 Reais
- Recife        51936.51 Reais
- Aracaju       48748.25 Reais
- Salvador      40596.73 Reais
- Fortaleza     37913.97 Reais

A loja que arrecadou mais foi a de ID 852 com um total de 527744.00 Reais, e a que menos arrecadou foi a de ID 1004 com um total de 4261.30 Reais